tensorboard/webapp/tbdev_upload/tbdev_upload_button_component.ts

```diff
@@ -32,15 +36,25 @@ import {TbdevUploadDialogComponent} from './tbdev_upload_dialog_component';
       .button-contents {
         align-items: center;
         display: flex;
+        text-transform: uppercase;
       }
       mat-icon {
         margin-right: 6px;
       }
     `,
   ],
 })
-export class TbdevUploadButtonComponent {
-  constructor(private dialog: MatDialog) {}
+export class TbdevUploadButtonComponent implements OnInit {
+  @HostBinding('class.shown') shown: boolean = false;
+
+  constructor(
+    @Inject('window') private window: Window,
+    private dialog: MatDialog
+  ) {}
+
+  ngOnInit() {
+    this.shown = LOCAL_HOSTNAMES.includes(this.window.location.hostname);
```

Comment:
this can be called in the constructor, I think. Does it need to happen in the OnInit? `OnInit` is the lifecycle method when you are guaranteed to have all `@Input()` passed but does not really buy you much.

Reply:
Yes it seems to work in the constructor. I thought I might need to use ngOnInit to ensure the reference to `window` works but I guess it makes sense that it should be available in the constructor.